leader-election-impl/src/main/java/com/palantir/paxos/SingleLeaderPinger.java

```diff
@@ -57,36 +60,78 @@
     private final boolean cancelRemainingCalls;
     private final Optional<OrderableSlsVersion> timeLockVersion;
     private final RateLimiter pingV2RateLimiter = RateLimiter.create(1.0 / (5 * 60));
-    private final RateLimiter greeningNodeShouldBecomeLeaderRateLimiter = RateLimiter.create(1.0 / (10 * 60));
+    private final GreenNodeLeadershipPrioritiser greenNodeLeadershipPrioritiser;
 
     private Map<LeaderPingerContext<PingableLeader>, Boolean> pingV2StatusOnRemotes = new HashMap<>();
 
-    public SingleLeaderPinger(
+    @VisibleForTesting
+    SingleLeaderPinger(
             Map<LeaderPingerContext<PingableLeader>, CheckedRejectionExecutorService> otherPingableExecutors,
             Duration leaderPingResponseWait,
             UUID localUuid,
             boolean cancelRemainingCalls,
-            Optional<OrderableSlsVersion> timeLockVersion) {
+            Optional<OrderableSlsVersion> timeLockVersion,
+            GreenNodeLeadershipPrioritiser greenNodeLeadershipPrioritiser) {
         this.leaderPingExecutors = otherPingableExecutors;
         this.leaderPingResponseWait = leaderPingResponseWait;
         this.localUuid = localUuid;
         this.cancelRemainingCalls = cancelRemainingCalls;
         this.timeLockVersion = timeLockVersion;
+        this.greenNodeLeadershipPrioritiser = greenNodeLeadershipPrioritiser;
+    }
+
+    // VisibleForTesting
+    public static SingleLeaderPinger createForTests(
+            Map<LeaderPingerContext<PingableLeader>, CheckedRejectionExecutorService> otherPingableExecutors,
+            Duration leaderPingResponseWait,
+            UUID localUuid,
+            boolean cancelRemainingCalls,
+            Optional<OrderableSlsVersion> timeLockVersion) {
+        GreenNodeLeadershipPrioritiser greenNodeLeadershipPrioritiser = new RateLimitedGreenNodeLeadershipPrioritiser();
+        return new SingleLeaderPinger(
+                otherPingableExecutors,
+                leaderPingResponseWait,
+                localUuid,
+                cancelRemainingCalls,
+                timeLockVersion,
+                greenNodeLeadershipPrioritiser);
+    }
+
+    public static SingleLeaderPinger create(
+            Map<LeaderPingerContext<PingableLeader>, CheckedRejectionExecutorService> otherPingableExecutors,
+            DataSource sqliteDataSource,
+            Duration leaderPingResponseWait,
+            Supplier<Duration> greenNodeLeadershipBackoff,
+            UUID localUuid,
+            boolean cancelRemainingCalls,
+            Optional<OrderableSlsVersion> timeLockVersion) {
+        GreenNodeLeadershipPrioritiser greenNodeLeadershipPrioritiser =
+                PersistedRateLimitingLeadershipPrioritiser.create(
```

Comment:
Are we using this one already? Feels like we should wait for the internal B/G config to be modified first - because otherwise we would be - in the worst case - only be soaking for half an hour.
Should we have a config flag for the type of prioritizer - seems excessive though

Reply:
We would be waiting a long time (maybe 3 months) before the internal B/G config to be ready - I don't really want to wait that long for this to be tested.

We could mitigate the worst case by adjusting the soak times (e.g. a constant 1 or 1.5 hours per node).

I also think that the worst-case (of the green node failing to gain leadership) should be rare, and that the very-worst-case (the green node failing to gain leadership *and* us not getting signal about some bug due to us not adjudicating on the green node for long enough) should be an order of magnitude or two rarer, especially given it would have to happen for each stack where we might have gotten signal.

I'm okay with merging with the new behaviour hard-disabled, provided we check with @jeremyk-91 if we still disagree as to whether this can be rolled out this year.

Reply:
IMO we should just be able to use the `PersistedRateLimitingLeadershipPrioritiser `  and kill `RateLimitedGreenNodeLeadershipPrioritiser` with `greenNodeLeadershipBackoffMs` configured to ~10min as it was originally. This will be strictly better than what we currently have.

Reply:
Cool!
At present RLGNLP is still easier to construct for tests.